metricproviders/prometheus/prometheus.go

```diff
@@ -162,21 +165,47 @@ func NewPrometheusAPI(metric v1alpha1.Metric) (v1.API, error) {
 	} else {
 		return nil, errors.New("prometheus address is not configured")
 	}
-	client, err := api.NewClient(api.Config{
-		Address: metric.Provider.Prometheus.Address,
-	})
-	if err != nil {
-		log.Errorf("Error in getting prometheus client: %v", err)
-		return nil, err
+
+	//Check if using Amazon Managed Prometheus if true build sigv4 client
+	promUrl := metric.Provider.Prometheus.Address
+	amznUrlSubstring := "aps-workspaces"
```

Comment:
that's not something expected to change. Would break too many customers.
But you maybe be more precise in the check based on https://docs.aws.amazon.com/general/latest/gr/prometheus-service.html

> dataplane endpoints are in the form aps-workspaces.region.amazonaws.com

Reply:
Alternatively, you could have the auth a configuration?

Reply:
Changed substring to more specific endpoint "aps-workspaces.region.amazonaws.com"